src/Features/Core/Portable/EmbeddedLanguages/StackFrame/StackFrameParser.cs

```diff
@@ -0,0 +1,455 @@
+            }
+
+            var identifier = _lexer.TryScanIdentifier();
+            if (!identifier.HasValue)
+            {
+                return ParseResult<StackFrameQualifiedNameNode>.Abort;
+            }
+
+            var (success, rhs) = TryScanGenericTypeIdentifier(identifier.Value);
+            if (!success)
+            {
+                return ParseResult<StackFrameQualifiedNameNode>.Abort;
+            }
+
+            RoslynDebug.AssertNotNull(rhs);
+            return new(new(lhs, dotToken, rhs));
+        }
+
+        /// <summary>
+        /// Given an identifier, attempts to parse the type identifier arity for it.
+        /// 
+        /// <code>
+        /// ex: MyNamespace.MyClass`1.MyMethod()
+        ///                 ^--------------------- MyClass would be the identifier passed in
+        ///                        ^-------------- Grave token
+        ///                         ^------------- Arity token of "1" 
+        /// </code>
+        /// 
+        /// </summary>
+        private ParseResult<StackFrameSimpleNameNode> TryScanGenericTypeIdentifier(StackFrameToken identifierToken)
+        {
+            if (!_lexer.ScanCurrentCharAsTokenIfMatch(StackFrameKind.GraveAccentToken, out var graveAccentToken))
+            {
+                return new(new StackFrameIdentifierNameNode(identifierToken));
+            }
+
+            var arity = _lexer.TryScanNumbers();
+            if (!arity.HasValue)
+            {
+                return ParseResult<StackFrameSimpleNameNode>.Abort;
+            }
+
+            return new(new StackFrameGenericNameNode(identifierToken, graveAccentToken, arity.Value));
+        }
+
+        /// <summary>
+        /// Type arguments for stacks are only valid on method declarations, and can have either '[' or '&lt;' as the 
+        /// starting character depending on output source.
+        /// 
+        /// ex: MyNamespace.MyClass.MyMethod[T](T t)
+        /// ex: MyNamespace.MyClass.MyMethod&lt;T&lt;(T t)
+        /// 
+        /// Assumes the identifier "MyMethod" has already been parsed, and the type arguments will need to be parsed.
+        /// </summary>
+        private ParseResult<StackFrameTypeArgumentList> TryParseTypeArguments()
+        {
+            if (!_lexer.ScanCurrentCharAsTokenIfMatch(
+                    kind => kind is StackFrameKind.OpenBracketToken or StackFrameKind.LessThanToken,
+                    out var openToken))
+            {
+                return ParseResult<StackFrameTypeArgumentList>.Empty;
+            }
+
+            var closeBrackeKind = openToken.Kind is StackFrameKind.OpenBracketToken
+                ? StackFrameKind.CloseBracketToken
+                : StackFrameKind.GreaterThanToken;
+
+            using var _ = ArrayBuilder<StackFrameNodeOrToken>.GetInstance(out var builder);
+            var currentIdentifier = _lexer.TryScanIdentifier(scanAtTrivia: false, scanLeadingWhitespace: true, scanTrailingWhitespace: true);
```

Comment:
tests with leading and trailing whitespace for these plz.

Reply:
Added